codegen-winter/src/air/boundary_constraints.rs

```diff
@@ -98,14 +98,14 @@ impl Codegen for BoundaryExpr {
         match self {
             Self::Const(value) => {
                 if is_aux_constraint {
-                    format!("E::from({})", value)
+                    format!("E::from({} as u64)", value)
                 } else {
-                    format!("Felt::from({})", value)
+                    format!("Felt::new({})", value)
                 }
             }
             Self::PubInput(name, index) => format!("self.{}[{}]", name, index),
             Self::Rand(index) => {
-                format!("aux_rand_elements[{}]", index)
+                format!("aux_rand_elements.get_segment_elements({})[0]", index)
```

Comment:
I think the index is being put in the wrong place here. First we want to get the random elements for the 1st aux segment (at index 0), then we want to index into them by `index`.

```
format!("aux_rand_elements.get_segment_elements(0)[{}]", index)
```

Reply:
Fixed, thanks.